packages/devtools_app/lib/src/app.dart

```diff
@@ -190,6 +195,21 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
     _clearCachedRoutes();
   }
 
+  /// Connects to the VM with the given URI.
+  ///
+  /// This request usually comes from the IDE via the server API to reuse the
+  /// DevTools window after being disconnected (for example if the user stops
+  /// a debug session then launches a new one).
+  void _connectVm(ConnectVmEvent event) {
+    routerDelegate.navigate(
+      homeScreenId,
```

Comment:
is there a reason we are navigating home now instead of just updating the args of the current screen like we were before?

Reply:
That wasn't triggering the VM service reconnection. I think there is probably a better way to do this (maybe as part of restructuring the router logic) but for now I just modified the `navigate` event to handle connecting to a VM service URI (along with disconnecting which it already handled). TLDR; using `navigate` so we connect to the vm service uri, and `navigate` requires a screen to navigate to

Reply:
ultimately `navigate` and `updateArgsIfChanged` both call `_replaceStack` and `notifyListeners`, so if we move the disconnect / connect logic to `_replaceStack` perhaps this is a better place.

Reply:
> , if I am on the embedded Inspector in VS Code and I disconnect from my current app, when I reconnect to a new app I want to still be on the inspector screen, not the home screen. CC @DanTup

Yep, but it's worth noting that this functionality is not for the embedded case. In the embedded case, we currently reconnect by just changing the URL for the iframe from the outside. This API is from pre-embedding (which you can still choose to use), where there's an external browser window (which VS Code can't change the URL for), so the messages goes via the server.